### ISSUE TITLE
ESM module: "Could not find a declaration file for module 'typeorm'" (0.3.6)

### ISSUE THREAD

**Issue by GuillaumeDesforges** [typeorm/typeorm#9075]
## Issue Description

A minimal ESM module with only TypeORM does not compile.

### Expected Behavior

The project should compile without error.

### Actual Behavior

```
$ yarn build
yarn run v1.22.18
warning package.json: No license field
$ tsc
src/index.ts:1:17 - error TS7016: Could not find a declaration file for module 'typeorm'. '/home/arsleust/perso/tmp/esm-module-ts-typeorm/node_modules/typeorm/index.mjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/typeorm` if it exists or add a new declaration (.d.ts) file containing `declare module 'typeorm';`

1 import orm from "typeorm"
                  ~~~~~~~~~


Found 1 error in src/index.ts:1

error Command failed with exit code 2.
```

```
// include the output in code tags like these!
```

### Steps to Reproduce

`package.json`
```json
{
  "name": "mypackage",
  "type": "module",
  "scripts": {
    "build": "tsc"
  },
  "devDependencies": {
    "typescript": "^4.8.0-dev.20220605"
  },
  "dependencies": {
    "typeorm": "^0.3.6"
  }
}
```

`tsconfig.json`
```
{
  "compilerOptions": {
    "target": "es2018",
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "esModuleInterop": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

`src/index.ts`
```ts
import orm from "typeorm"

```

Run `yarn build`

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Node.js version     | 16.14.2  | 
| Typescript version  | 4.8.0-dev.20220605  | 
| TypeORM version     | 0.3.6  | 

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ Yes, I have the time, and I know how to start.
 - ✅ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by sourcecodes2** [comment]
[DemianD](https://github.com/DemianD), thank you for your fix!

**Comment by AlexMesser** [comment]
fixed in #9097
